server/src/app/setup/index.ts

```diff
@@ -0,0 +1,8 @@
+export { Entities } from './entities'
+export { MikroORM } from './mikro-orm'
+export { MikroEntityRepos } from './mikro-entity-repos'
+export { MikroRepos } from './mikro-repos'
```

Comment:
This opens the door for us to do `/database`, `/application`, and `/http`, so I'd be down to do that kind of grouping

Reply:
Yeah I agree

Reply:
This made me realize that Waterpark setup actually only has three stages.

1. **db** - There are two ways to set up the DB: mikro (`setupMikroDB`) or mock (`setupMockDB`).
2. **application** - There is only one way to set up the application using our `UseCases` and `Controllers`, which is the `setupApplication` function.
3. **http** - For now, we can only choose to use Express. There are infinitely many ways we could choose the Waterpark Express API handlers. However, we only use one that features a versioned API: `setupWaterparkExpress` (we might add others in the future).

Each stage might have smaller steps within, but we can only really make meaningful decisions at the stage level. For example, it doesn't make sense to micromanage how Mikro goes from `Entity`s to `EntityRepo`s to `Repo`s. We might as well let a single factory function (`setupMikroDB`) handle all of it. What we do want to choose is whether we want a Mikro-based DB, or just a mock DB.

Just like `mockCreateUser`, the stage-level setup functions return all the intermediate objects in case the caller wants to mess with them.

Reply:
@xujustinj Great idea! 🔥